code/modules/projectiles/guns/projectile/pistol/lamia.dm

```diff
@@ -1,123 +1,122 @@
-	init_recoil = HANDGUN_RECOIL(1.1)
-	unload_sound 	= 'sound/weapons/guns/interact/avasarala_unload.ogg'
-	reload_sound 	= 'sound/weapons/guns/interact/hpistol_magin.ogg'
-	cocked_sound 	= 'sound/weapons/guns/interact/avasarala_cock.ogg'
-	fire_sound 	= 'sound/weapons/guns/fire/avasarala_fire.ogg'
-	serial_type = "NM"
-	gun_parts = list(/obj/item/part/gun/frame/amnesty = 1, /obj/item/part/gun/grip/rubber = 1, /obj/item/part/gun/mechanism/pistol = 1, /obj/item/part/gun/barrel/kurtz = 1)
-
-/obj/item/part/gun/frame/amnesty
-	name = "Amnesty frame"
-	desc = "An Amnesty pistol frame. For a pocket sized hole puncher."
-	icon_state = "frame_lamia"
-	result = /obj/item/gun/projectile/lamia/amnesty
-	resultvars = list(/obj/item/gun/projectile/lamia/amnesty)
-	gripvars = list(/obj/item/part/gun/grip/rubber)
-	mechanismvar = /obj/item/part/gun/mechanism/pistol
-	barrelvars = list(/obj/item/part/gun/barrel/kurtz)
+/obj/item/gun/projectile/lamia
+	name = "\"Lamia\" heavy pistol"
+	desc = "A large, unwieldy handgun based on the Ladon design. Chambered in 12mm, made for officers and heavy units."
+	icon = 'icons/obj/guns/projectile/lamia.dmi'
+	icon_state = "lamia"
+	item_state = "lamia"
+	fire_sound = 'sound/weapons/guns/fire/hpistol_fire.ogg'
+	caliber = CAL_50
+	origin_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 4)
+	can_dual = TRUE
+	load_method = SINGLE_CASING|MAGAZINE
+	mag_well = MAG_WELL_PISTOL
+	matter = list(MATERIAL_PLASTEEL = 15, MATERIAL_PLASTIC = 8)
+	price_tag = 1000
+	unload_sound 	= 'sound/weapons/guns/interact/hpistol_magout.ogg'
+	reload_sound 	= 'sound/weapons/guns/interact/hpistol_magin.ogg'
+	cocked_sound 	= 'sound/weapons/guns/interact/hpistol_cock.ogg'
+	damage_multiplier = 1.2
+	penetration_multiplier = 1.3
+	init_recoil = HANDGUN_RECOIL(1.3)
+	gun_tags = list(GUN_PROJECTILE, GUN_MAGWELL, GUN_CALIBRE_12MM)
+	serial_type = "H&S"
+
+	wield_delay = 0.6 SECOND
+	wield_delay_factor = 0.6 // 60 vig
+	gun_parts = list(/obj/item/part/gun/frame/lamia = 1, /obj/item/part/gun/grip/rubber = 1, /obj/item/part/gun/mechanism/pistol = 1, /obj/item/part/gun/barrel/kurtz = 1)
+
+/obj/item/part/gun/frame/lamia
+	name = "Lamia frame"
+	desc = "An Lamia pistol frame. For a pocket sized hole puncher."
+	icon_state = "frame_lamia"
+	result = /obj/item/gun/projectile/lamia
+	resultvars = list(/obj/item/gun/projectile/lamia)
+	gripvars = list(/obj/item/part/gun/grip/rubber)
+	mechanismvar = /obj/item/part/gun/mechanism/pistol
+	barrelvars = list(/obj/item/part/gun/barrel/kurtz)
+
+/obj/item/gun/projectile/lamia/update_icon()
+	..()
+
+	if(!ammo_magazine)
+		icon_state = initial(icon_state)
+	else if(!ammo_magazine.stored_ammo.len)
+		icon_state = initial(icon_state) + "_empty"
+	else
+		icon_state = initial(icon_state) + "_full"
+
+/obj/item/gun/projectile/lamia/gemini
+	name = "\"Gemini\" advanced heavy pistol"
+	desc = "An upgraded Lamia pistol fitted with a smart scope that harmonizes with neural link software. Chambered in 12mm, with an underbarrel LAM."

```

Comment:
What's a LAM?

Reply:
Standard short term for "Laser Aiming Module"